Python/fileutils.c

```diff
@@ -2096,17 +2107,131 @@ _Py_abspath(const wchar_t *path, wchar_t **abspath_p)
 #endif
 }
 
+// The Windows Games API family does not provide these functions
+// so provide our own implementations. Remove them in case they get added
+// to the Games API family
+#if defined(MS_WINDOWS) && !defined(MS_WINDOWS_APP) && !defined(MS_WINDOWS_SYSTEM)
+HRESULT
+PathCchSkipRoot(const wchar_t *path, const wchar_t **rootEnd)
+{
+    if (path[0] == '\\') {
+        /* relative path with root e.g. \Windows */
+        if (path[1] != '\\') {
+            *rootEnd = path + 1;
+            return S_OK;
+        }
+
+        /* UNC drives e.g. \\server\share or \\?\UNC\server\share */
+        const wchar_t *end = path + 2;
+        if (!wcsnicmp(end, L"?\\UNC\\", 6)) {
+            end += 6;
+        }
+
+        end = wcschr(end, '\\');
+        if (!end) {
+            *rootEnd = path + wcslen(path);
+            return S_OK;
+        }
+        end = wcschr(end + 1, '\\');
+        *rootEnd = (!end) ? path + wcslen(path) : end + 1;
+        return S_OK;
+    }
+    /* absolute / relative path with drive, e.g. C: or C:\ */
+    else if (isalpha(path[0]) && path[1] == ':') {
+        *rootEnd = (path[2] == '\\') ? path + 3 : path + 2;
+        return S_OK;
+    }
+
+    /* relative path */
+    return E_INVALIDARG;
+}
+
+static HRESULT
+PathCchStripToRoot(wchar_t *path, size_t size)
+{
+    wchar_t *end;
+    if (PathCchSkipRoot(path, &end) == S_OK) {
+        if (*end == '\0') {
+            return S_FALSE;
+        }
+        *end = '\0';
+    }
+
+    return E_INVALIDARG;
+}
+
+static wchar_t*
+PathAddBackslashW(wchar_t *path)
+{
+    size_t len;
+    if (!path) {
+        return NULL;
+    }
+    len = wcslen(path);
+    if (len && path[len - 1] != '\\') {
+        path[len++] = '\\';
+        path[len] = '\0';
+    }
+    return path + len;
+}
+
+static int
+win32_games_join_relfile(wchar_t *buffer, size_t bufsize,
```

Comment:
Is there a reason to implement this as `join_relfile` instead of `PathCchCombineEx`? We don't really have to support the flags for it.

Reply:
I mainly did not rename it, since I do not provide all features:
- no normalization
- no flags support
- not the whole LocalAlloc stuff

Reply:
Now that there is a comment, that they do not behave the same I guess this is no issue